### PR TITLE
Adjust `to_edisp_kernel` input to `MapAxis`

### LINKED ISSUE #5142: Expect the `MapAxis` directly in `to_edisp_kernel`
Should we expect directly the `MapAxis` here? (Note: this is a TODO converted into an issue)

https://github.com/gammapy/gammapy/blob/e5aecb334e7aebe304affd96b545a636019f7626/gammapy/irf/edisp/core.py#L145-L148

**Solution**
'energy' should be a `MapAxis` type. This needs a small API changes and deprecations so that 'energy' is a `MapAxis` and not a `Quantity`.

### PR DESCRIPTION
**Pull request by Astro-Kirsty**
This PR is to resolve #5142. 
I left it as a draft as I am not sure if this is the correct procedure to take. 
This is two part issue, we want to change the variables to be consistent with the other naming for axis https://github.com/gammapy/gammapy/blob/86488ece61f41e92ff3212b9b870f5a9f7d68b79/gammapy/irf/edisp/map.py#L116
We also need to change the variable to no longer be a quantity but a `MapAxis`. 

This was the best way I could think to achieve this.

Note: there are a number of failing tests, but I did not want to fix them all with the deprecation warning until a method was decided upon.

I am open to suggestions.

### CODE REVIEW

File: gammapy/irf/edisp/core.py [hunk 1]
```diff
@@ -117,7 +124,13 @@ def from_gauss(
             data=data.value,
         )
 
-    def to_edisp_kernel(self, offset, energy_true=None, energy=None):
+    @deprecated_renamed_argument(
+        ["energy_true", "energy"],
+        ["energy_axis_true", "energy_axis"],
```
Comment (AtreyeeS) [inline gammapy/irf/edisp/core.py:129]: Do you want to rename the argument name or keep the name and change the type?
Either of the two should suffice, no?

If a `DepricationWarning` is raised about the argument name, then the `warning` below is not necessary.

I think we can keep the argument same same, and just allow a `MapAxis` on input. Then, we can raise a deprecation warning if a Quantity is passed

**Review by AtreyeeS — CHANGES_REQUESTED** [review]
Thanks @Astro-Kirsty ! 

I do not think you need to adapt internet tests to check for the deprecation warning, eg as done in `gammapy/datasets/tests/test_map.py` 
It only needs to be checked when the user passes a quantity instead of a `MapAxis` in the `to_edisp_kernel` function, and internally within Gammapy we should pass a `MapAxis`.

File: gammapy/irf/edisp/core.py (hunk 1 above)
Comment (Astro-Kirsty) [reply gammapy/irf/edisp/core.py:129]: I changed the argument name to be consistent with others in the code, for example [here](https://github.com/gammapy/gammapy/blob/18b65d5cce21876d53aa7faf54aaeccd68e8b76c/gammapy/irf/edisp/map.py#L116).

Comment (Astro-Kirsty) [reply gammapy/irf/edisp/core.py:129]: Okay sure thing! Will do

Comment (Astro-Kirsty) [reply gammapy/irf/edisp/core.py:129]: So `deprecated_renamed_argument` will only let the user know that there was a name change, but not that it should now be a `MapAxis`

**Review by QRemy — DISMISSED** [review]
Thanks @Astro-Kirsty looks good.

**Review by AtreyeeS — APPROVED** [review]
Thanks @Astro-Kirsty ! No further comments

File: gammapy/irf/edisp/core.py [hunk 2]
```diff
@@ -141,18 +153,11 @@ def to_edisp_kernel(self, offset, energy_true=None, energy=None):
 
         offset = Angle(offset)
 
-        if energy is None:
+        if energy_axis is None:
```
Comment (AtreyeeS) [inline gammapy/irf/edisp/core.py:156]: Will this work correctly for the old behaviour? We have to support passing quantities for 1.3

**Review by AtreyeeS — COMMENTED** [review]
Please check that the deprecation works correctly, ie, the old behaviour is still supported

File: gammapy/irf/edisp/core.py (hunk 2 above)
Comment (Astro-Kirsty) [reply gammapy/irf/edisp/core.py:156]: Apologies I removed a few lines when removed the warning.
Fixing now.

**Review by AtreyeeS — APPROVED** [review]
No further comments. Thanks!

**Review by registerrier — APPROVED** [review]
Thanks @Astro-Kirsty . This looks good!

**Pull request merged**